test/server/cards/06_SEC/leaders/ColonelYularenThisIsWhyWePlan.spec.ts

```diff
@@ -0,0 +1,201 @@
+                        leader: { card: 'colonel-yularen#this-is-why-we-plan', deployed: true }
+                    },
+                    player2: {
+                        groundArena: ['sundari-peacekeeper'],
+                        spaceArena: ['tie-advanced']
+                    },
+                });
+            });
+
+            it('should allow attacking with another unit that costs 4 or less', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.colonelYularenThisIsWhyWePlan);
+                context.player1.clickCard(context.sundariPeacekeeper);
+
+                expect(context.colonelYularenThisIsWhyWePlan.exhausted).toBe(true);
+                expect(context.colonelYularenThisIsWhyWePlan.damage).toBe(1);
+                expect(context.sundariPeacekeeper.damage).toBe(4);
+
+                expect(context.player1).toBeAbleToSelectExactly([context.sabineWrenExplosivesArtist, context.deathTrooper, context.tielnFighter, context.seventhFleetDefender]);
+                expect(context.player1).toHavePassAbilityButton();
+
+                context.player1.clickCard(context.deathTrooper);
+                expect(context.player1).toHavePassAttackButton();
+                context.player1.clickCard(context.p2Base);
+                expect(context.deathTrooper.exhausted).toBe(true);
+                expect(context.p2Base.damage).toBe(3);
+                expect(context.deathTrooper.damage).toBe(0);
+
+                context.player2.passAction();
+
+                // second Yularen attack to confirm that passing the ability works
+                context.readyCard(context.colonelYularenThisIsWhyWePlan);
+                context.player1.clickCard(context.colonelYularenThisIsWhyWePlan);
+                context.player1.clickCard(context.p2Base);
+                expect(context.colonelYularenThisIsWhyWePlan.exhausted).toBe(true);
+                expect(context.p2Base.damage).toBe(7);
+
+                expect(context.player1).toHaveEnabledPromptButton('Pass');
+                context.player1.clickPrompt('Pass');
+
+                context.player2.passAction();
+
+                // third Leia attack to confirm that the ability isn't triggered if there is no legal attacker
+                context.exhaustCard(context.sabineWrenExplosivesArtist);
+                context.exhaustCard(context.tielnFighter);
+                context.exhaustCard(context.seventhFleetDefender);
+                context.readyCard(context.colonelYularenThisIsWhyWePlan);
+                context.player1.clickCard(context.colonelYularenThisIsWhyWePlan);
+                context.player1.clickCard(context.p2Base);
+                expect(context.colonelYularenThisIsWhyWePlan.exhausted).toBe(true);
+                expect(context.p2Base.damage).toBe(11);
+
+                expect(context.player2).toBeActivePlayer();
+            });
```

Comment:
can you add some test when yularen die on his attack (against an atst for example) or with new ahsoka (ahsoka trigger should be done in first)